Added French language

Mesaan opened the pull request.

Mesaan committed:
Added French Language

Mesaan committed:
Minor language fix

furkansimsekli commented:
Hey there, huge thanks for the great work! I just had the chance to test it. Some letters you used in the department names are missing in the list of `CHARS`. That's why sorting function doesn't work.

## (')
I know it's not a member of the alphabet in French, but as I did for the whitespace, you need to do something similar here. `CHARS` is not just a list of letters but it also represents the priority. I tested by adding it at the end of the list and it worked, but not sure if it's correct.

Let's look at this two example, is this correct order? or should it be reversed?

1. Département de Mathématiques
2. Département d'Ingéniérie Industrielle

## (ô) and (Ô)
I also saw that `ô` is not in the list. Afaik that's not an actually letter from alphabet but it's like an accent or something? Turkish has also similar thing with the `a`, but it's not something common. Maybe we can place it after the usual `O` and `o`.

furkansimsekli committed:
locale.fr: add accents and symbol '

I didn't like adding a symbol here. I thought maybe there would be
a list of fixed symbols and I would add them at the beginning of
CHARS list before sorting. Then non-latin languages could be a problem
because I can't control which language has which special symbols.

Let's keep this way for now, add every letter that are used for the
department names into the CHARS. If anything comes up later, I can
consider finding a different solution, maybe using a 3rd party library.
I should also look how they solve this issue, I mean many applications
nowadays have multilanguage support. They must have an sorting rule
for each of them.
